File: Android/src/androidMain/kotlin/com/taskbridge/android/repository/TasksRepository.kt
package com.taskbridge.android.repository

import com.taskbridge.core.models.tasks.TaskId
import com.taskbridge.core.models.tasks.TaskItem
import com.taskbridge.core.usecases.tasks.TasksState
import kotlinx.coroutines.flow.Flow

interface TasksRepository {
    val tasksState: Flow<TasksState>
    suspend fun loadTasks()
    suspend fun createTask(task: TaskItem)
    suspend fun replaceTask(task: TaskItem)
    suspend fun deleteTaskTree(taskId: TaskId)
}

File: iOS/TaskBridge/Repository/TasksRepository.swift
import Foundation
import TaskBridgeCore

@MainActor
protocol TasksRepository {
    var tasksState: AsyncStream<TasksState> { get }
    
    func loadTasks() async throws
    func createTask(_ task: TaskItem) async throws
    func replaceTask(_ task: TaskItem) async throws
    func deleteTaskTree(_ taskId: TaskId) async throws
}
